### PR TITLE
feat: add support for mariadb datasource, 

### PR DESCRIPTION
**Pull request by happenedIn**
for #19 added unit test, added constants file for database product names

### CODE REVIEW

**Comment by hsluoyz** [comment]
@tangyang9464 @imp2002 please review

File: src/main/java/org/casbin/constants/AdapterConstants.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.casbin.constants;
+
+public enum AdapterConstants {
```
Comment (hsluoyz) [inline src/main/java/org/casbin/constants/AdapterConstants.java:3]: This code is too tedius

File: src/main/java/org/casbin/adapter/HibernateAdapter.java [hunk 2]
```diff
@@ -59,10 +60,10 @@ private void open() throws SQLException {
     private void createDatabase() {
         Session session = factory.openSession();
         Transaction tx = session.beginTransaction();
-        if (this.databaseProductName.contains("MySQL")) {
+        if (this.databaseProductName.contains(AdapterConstants.MySQL.name()) || this.databaseProductName.contains(AdapterConstants.MariaDB.name())) {
```
Comment (hsluoyz) [inline src/main/java/org/casbin/adapter/HibernateAdapter.java:63]: Don't repeat this thing so many times. Make it simpler and smaller

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
Closed for low code quality

File: src/main/java/org/casbin/adapter/HibernateAdapter.java (hunk 2 above)
Comment (happenedIn) [reply src/main/java/org/casbin/adapter/HibernateAdapter.java:63]: thanks, added a new PR that is simpler

**Pull request closed without merging**